jacodb-analysis/src/main/kotlin/org/jacodb/analysis/util/EtsTraits.kt

```diff
@@ -0,0 +1,160 @@
+import org.jacodb.panda.dynamic.ets.base.EtsStmt
+import org.jacodb.panda.dynamic.ets.base.EtsThis
+import org.jacodb.panda.dynamic.ets.base.EtsValue
+import org.jacodb.panda.dynamic.ets.model.EtsMethod
+import org.jacodb.panda.dynamic.ets.model.EtsMethodImpl
+import org.jacodb.panda.dynamic.ets.model.EtsMethodParameter
+import org.jacodb.panda.dynamic.ets.utils.callExpr
+import org.jacodb.taint.configuration.ConstantValue
+import org.jacodb.analysis.util.toPath as _toPath
+import org.jacodb.analysis.util.toPathOrNull as _toPathOrNull
+import org.jacodb.panda.dynamic.ets.utils.getOperands as _getOperands
+import org.jacodb.panda.dynamic.ets.utils.getValues as _getValues
+
+interface EtsTraits : Traits<EtsMethod, EtsStmt> {
+
+    override val CommonCallExpr.callee: EtsMethod
+        get() {
+            check(this is EtsCallExpr)
+            // return cp.getMethodBySignature(method) ?: error("Method not found: $method")
+            return EtsMethodImpl(method)
+        }
+
+    override val EtsMethod.thisInstance: EtsThis
+        get() = EtsThis(EtsAnyType)
+
+    override val EtsMethod.isConstructor: Boolean
+        get() = false
+
+    override fun CommonExpr.toPathOrNull(): AccessPath? {
+        check(this is EtsEntity)
+        return this._toPathOrNull()
+    }
+
+    override fun CommonValue.toPathOrNull(): AccessPath? {
+        check(this is EtsValue)
+        return this._toPathOrNull()
+    }
+
+    override fun CommonValue.toPath(): AccessPath {
+        check(this is EtsValue)
+        return this._toPath()
+    }
+
+    override fun getArgument(param: CommonMethodParameter): EtsParameterRef {
+        check(param is EtsMethodParameter)
+        return EtsParameterRef(index = param.index, type = param.type)
+    }
+
+    override fun getArgumentsOf(method: EtsMethod): List<CommonArgument> {
+        return method.parameters.map { getArgument(it) }
+    }
+
+    override fun CommonValue.isConstant(): Boolean {
+        check(this is EtsEntity)
+        return this is EtsConstant
+    }
+
+    override fun CommonValue.eqConstant(constant: ConstantValue): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    override fun CommonValue.ltConstant(constant: ConstantValue): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    override fun CommonValue.gtConstant(constant: ConstantValue): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    override fun CommonValue.matches(pattern: String): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    override fun EtsStmt.getCallExpr(): EtsCallExpr? {
+        return callExpr
+    }
+
+    override fun CommonExpr.getValues(): Set<EtsValue> {
+        check(this is EtsEntity)
+        return _getValues().toSet()
+    }
+
+    override fun EtsStmt.getOperands(): List<EtsEntity> {
+        return _getOperands().toList()
+    }
+
+    companion object : EtsTraits {
+        // Note: unused for now
+        // lateinit var cp: EtsFile
```

Comment:
We do not need it *right now*, but it *might` be useful in the future for methods similar to `getArguments` in JcTraits. Instead of making such methods/properties to be extensions of CommonProject (for example, if your property already has some more important receiver), you either convert it to method argument (downside is that you cannot use property style anymore, and you have to pass extra arg everywhere), or you just *have* cp "in the context" -- in the Traits instance.

Current solution for `lateinit var cp: JcClasspath` in `JcTraits` should be replaced with the proper `data class JcTraits(val cp: JcClasspath)`. Similar things could be done for EtsTraits in the futute, when there will be a necessity to have cp in the context.
I am aware of it and try to fix it in nearest time.